packages/datatrak-web/src/features/Survey/Screens/SurveySuccessScreen.tsx

```diff
@@ -69,13 +70,25 @@ const Button = styled(BaseButton)`
   }
 `;
 
+const ReturnButton = () => {
+  const from = useFromLocation();
+  return from === ROUTES.TASKS ? (
```

Comment:
Since we are going to be implementing this on the task details page as well, what do you think of making the page name either linked to the route somewhere, or saving it in the location state as well? Then we don't need any ternaries